### ISSUE TITLE
Omitting access_key and secret_key in Alertmanager SNS configuration causes a crash

### ISSUE THREAD

**Issue by mlaitinen** [prometheus-operator/prometheus-operator#5130]
**What happened?**

I configured Alertmanager with an SNS receiver and defined a sigv4 configuration without `access_key` and `secret_key`. This causes Prometheus Operator to crash.

I have to define the sigv4 configuration, because otherwise I get this:
```
caller=notify.go:732 level=warn component=dispatcher receiver=prometheus/sns-email/sns-email-mycompany integration=sns[0] msg="Notify attempt failed, will retry later" attempts=1 err="region not configured in sns.sigv4.region or in default credentials chain"
```

**Did you expect to see something different?**

As mentioned in [Prometheus documentation](https://prometheus.io/docs/alerting/latest/configuration/#sigv4_config), `role_arn` can be used as "an alternative to using AWS API keys". I expected Prometheus Operator to ignore `access_key` and `secret_key`, because I had `role_arn` and `region` defined.

**How to reproduce it (as minimally and precisely as possible)**:

Create an `AlertmanagerConfig` manifest, define a `sigv4` config with `region` and `role_arn`, but omit `access_key` and `secret_key`. View Prometheus Operator logs.

**Environment**

* Prometheus Operator version:

    `v0.59.1`
    
    

* Kubernetes version information:

```
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.3", GitCommit:"816c97ab8cff8a1c72eccca1026f7820e93e0d25", GitTreeState:"clean", BuildDate:"2022-01-25T21:25:17Z", GoVersion:"go1.17.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"23+", GitVersion:"v1.23.10-eks-15b7512", GitCommit:"cd6399691d9b1fed9ec20c9c5e82f5993c3f42cb", GitTreeState:"clean", BuildDate:"2022-08-31T19:17:01Z", GoVersion:"go1.17.13", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind:

    EKS, managed by Terraform

* Manifests:

```
apiVersion: monitoring.coreos.com/v1alpha1
kind: AlertmanagerConfig
metadata:
  labels:
    alertmanagerConfig: sns-email
spec:
  receivers:
    - name: sns-email-mycompany
      snsConfigs:
        - sendResolved: true
          sigv4:
            region: eu-west-1
            roleArn: >-
              arn:aws:iam::123456789:role/test-eks-role-eksClusterRole-YXVEZ3L123456
          topicARN: arn:aws:sns:eu-west-1:123456789:cluster-alerts
  route:
    groupBy:
      - job
    groupInterval: 5m
    groupWait: 30s
    receiver: sns-email-mycompany
    repeatInterval: 12h
    routes:
      - matchers:
          - matchType: '!='
            name: alertname
            value: KubeHpaMaxedOut
          - matchType: '!='
            name: alertname
            value: KubeControllerManagerDown
          - matchType: '!='
            name: alertname
            value: KubeSchedulerDown
        receiver: sns-email-mycompany

```

* Prometheus Operator Logs:

```
level=info ts=2022-11-01T13:05:13.330287186Z caller=operator.go:320 component=thanosoperator msg="CRD API endpoints ready"
level=info ts=2022-11-01T13:05:13.43046244Z caller=operator.go:263 component=thanosoperator msg="successfully synced all caches"
level=info ts=2022-11-01T13:05:13.631238297Z caller=operator.go:288 component=alertmanageroperator msg="successfully synced all caches"
level=info ts=2022-11-01T13:05:13.631234623Z caller=operator.go:369 component=prometheusoperator msg="successfully synced all caches"
level=info ts=2022-11-01T13:05:13.631468752Z caller=operator.go:753 component=alertmanageroperator key=prometheus/prometheus-kube-prometheus-alertmanager msg="sync alertmanager"
level=info ts=2022-11-01T13:05:13.631607093Z caller=operator.go:1559 component=prometheusoperator key=prometheus/prometheus-kube-prometheus-prometheus msg="update prometheus status"
level=info ts=2022-11-01T13:05:13.631955139Z caller=operator.go:1390 component=prometheusoperator key=prometheus/prometheus-kube-prometheus-prometheus msg="sync prometheus"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1908171]

goroutine 43 [running]:
github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*configBuilder).convertSnsConfig(0xc0011c6590, {0x2293b78, 0xc0002cf600}, {0xc0011ba0a9, {0x0, 0x0}, 0xc000737380, {0xc0007f8400, 0x31}, {0x0, ...}, ...}, ...)
	/home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/amcfg.go:1052 +0x2f1
github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*configBuilder).convertReceiver(0xc0002afa90?, {0x2293b78, 0xc0002cf600}, 0xc0011c6270, {{0xc0011ba0b0?, 0x0?}, {0xc0011ba0a0?, 0x227d7e0?}})
	/home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/amcfg.go:525 +0x17ea
github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*configBuilder).initializeFromAlertmanagerConfig(0xc0011c6590, {0x2293b78, 0xc0002cf600}, 0x0?, 0xc0008cc580)
	/home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/amcfg.go:148 +0x492
github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*Operator).provisionAlertmanagerConfiguration(0xc000102ea0, {0x2293b78, 0xc0002cf600}, 0xc0002da400, 0xc00039b860)
	/home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/operator.go:977 +0x4c5
github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*Operator).sync(0xc000102ea0, {0x2293b78, 0xc0002cf600}, {0xc000053180, 0x32})
	/home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/operator.go:757 +0x5c5
github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*Operator).processNextWorkItem(0xc000102ea0, {0x2293b78, 0xc0002cf600})
	/home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/operator.go:548 +0x167
github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*Operator).worker(0x8f08aa?, {0x2293b78, 0xc0002cf600})
	/home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/operator.go:535 +0x39
created by github.com/prometheus-operator/prometheus-operator/pkg/alertmanager.(*Operator).Run
	/home/runner/work/prometheus-operator/prometheus-operator/pkg/alertmanager/operator.go:470 +0x30a
```

**Anything else we need to know?**:
I tried defining `access_key` and `secret_key` as empty strings, but that doesn't work.

**Comment by slashpai** [comment]
Thanks for reporting the issue!

**Comment by PhilipGough** [comment]
Closed via #5143
